packages/protocol/contracts/rrp/requesters/RrpRequester.sol

```diff
@@ -15,8 +15,13 @@ contract RrpRequester {
     }
 
     /// @dev Airnode RRP address is set at deployment and is immutable
+    /// The Rrp Requester is set at its own sponsor
     /// @param airnodeRrpAddress Airnode RRP contract address
     constructor(address airnodeRrpAddress) {
         airnodeRrp = IAirnodeRrp(airnodeRrpAddress);
+        IAirnodeRrp(airnodeRrpAddress).setSponsorshipStatus(
```

Comment:
Cannot directly reference airnodeRrp because of the immutable nature of airnodeRrp 
> Immutable variables cannot be read during contract creation time, which means they cannot be read in the constructor or any function or modifier called from it.

Reply:
Interesting 🤔 tho it could be called if you create a local copy of airnodeRrp and call `setSponsorshipStatus()` using that copy.

Reply:
yup that is also possible but I thought this was a cleaner approach.

Reply:
Yeah the current implementation is good